src/iterators/kruskal.jl

```diff
@@ -0,0 +1,46 @@
+# The kruskal algorithms are largely copied from Graphs/src/spanningtrees/kruskal.jl
+
+struct KruskalIterator <: EdgeIterator
+    graph::AbstractGraph
+    connected_vs::IntDisjointSets
+    distmx::AbstractMatrix
+    edge_list
+
+    function KruskalIterator(graph, distmx=weights(g); minimize=true)
+        is_directed(graph) && throw(ArgumentError("Cannot use Kruskal on a directed graph."))
+        weights = Vector{eltype(distmx)}()
+        sizehint!(weights, ne(graph))
+        edge_list = collect(edges(graph))
+        for e in edge_list
+            push!(weights, distmx[src(e), dst(e)])
+        end
+        e = edge_list[sortperm(weights; rev=!minimize)]
+        new(graph, IntDisjointSets(nv(graph)), distmx, e)
+    end
+end
+
+Base.length(t::KruskalIterator) = nv(t.graph)-1
```

Comment:
Is that true in case the graph is not connected?

Reply:
I think you are completely correct, but I don't have time to work on this at the moment, instead I would be willing to finish up #163, if you happen to have some review time I think it could be better to redirect it there :-)

(by the way I think also the other PR suffer from the same problem)